### PR TITLE
fix: P-441 remove NoEligibleIdentity

### PR DESCRIPTION
**Pull request by wli-pro**
return false value VC if has no eligible identity

### Labels

Please apply following PR-related labels when appropriate:
- `C0-breaking`: if your change could break the existing client, e.g. API change, critical logic change 
- `C1-noteworthy`: if your change is non-breaking, but is still worth noticing for the client, e.g. reference code improvement

### Testing Evidences

Please attach any relevant evidences if applicable

### CODE REVIEW

File: tee-worker/client-api/parachain-api/prepare-build/interfaces/sidechain/definitions.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export default {
                 UnexpectedMessage: "Null",
                 __Unused_WrongSignatureType: "Null",
                 VerifyWeb3SignatureFailed: "Null",
-                NoEligibleIdentity: "Null",
+                __Unused_NoEligibleIdentity: "Null",
```
Comment (Kailai-Wang) [inline tee-worker/client-api/parachain-api/prepare-build/interfaces/sidechain/definitions.ts]: I suggest removing it completely.

Imagine we have a new entry later, we might erroneously append them like:
```
// in rust
#[codec(index = 11)]
NewEntry,

// in definitions.ts
__Unused_NoEligibleIdentity: "Null",
NewEntry: "Null",
```

Which is wrong, so it's very error-prone IMO.

**Review by jonalvarezz — APPROVED** [review]
thanks!

File: tee-worker/app-libs/stf/src/trusted_call_litentry.rs [hunk 2]
```diff
@@ -196,10 +196,9 @@ impl TrustedCallSigned {
 		let assertion_networks = assertion.get_supported_web3networks();
 		let identities = get_eligible_identities(id_graph.as_ref(), assertion_networks);
 
-		ensure!(
-			!identities.is_empty(),
-			StfError::RequestVCFailed(assertion, ErrorDetail::NoEligibleIdentity)
-		);
+		if identities.is_empty() {
+			warn!("[RequestVc] : No eligible identities");
```
Comment (BillyWooo) [inline tee-worker/app-libs/stf/src/trusted_call_litentry.rs]: if the purpose is to not leak user info, should here still print out such text?

File: tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs [hunk 3]
```diff
@@ -46,7 +46,12 @@ pub fn build(
 	})?;
 
 	if response.is_empty() {
-		Err(Error::RequestVCFailed(Assertion::BRC20AmountHolder, ErrorDetail::NoEligibleIdentity))
+		Err(Error::RequestVCFailed(
+			Assertion::BRC20AmountHolder,
+			ErrorDetail::StfError(ErrorString::truncate_from(
+				"BRC20 token not found for linked identity".as_bytes().to_vec(),
+			)),
```
Comment (BillyWooo) [inline tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs]: same here. 
Do I understand it right: even there is no eligible identity, an empty/false VC will still be created, right? Here will just return error message. No empty/dummy VC.

File: tee-worker/litentry/core/vc-task/receiver/src/lib.rs [hunk 4]
```diff
@@ -529,7 +529,10 @@ where
 		let id_graph_hash = H256::from(blake2_256(&id_graph.encode()));
 		let assertion_networks = assertion.get_supported_web3networks();
 		let identities = get_eligible_identities(id_graph.as_ref(), assertion_networks);
-		ensure!(!identities.is_empty(), "No eligible identity".to_string());
+
+		if identities.is_empty() {
+			warn!("[RequestVc] : No eligible identities");
```
Comment (BillyWooo) [inline tee-worker/litentry/core/vc-task/receiver/src/lib.rs]: same here.

**Review by BillyWooo — COMMENTED** [review]
In general, if we are going to create dummy/false value VC, we need to apply this change to all VCs. 
Or as we are moving to solidity, should we continue on this topic?

File: tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs (hunk 3 above)
Comment (wli-pro) [reply tee-worker/litentry/core/assertion-build/src/brc20/amount_holder.rs]: Fixed in this commit now: https://github.com/litentry/litentry-parachain/pull/2739/commits/9bf130b8b44100863a9ff89d3c5502bd2ceef238

**Comment by wli-pro** [comment]
> In general, if we are going to create dummy/false value VC, we need to apply this change to all VCs. Or as we are moving to solidity, should we continue on this topic?

All VCs are being applied with this same handling.

File: tee-worker/litentry/core/vc-task/receiver/src/lib.rs (hunk 4 above)
Comment (wli-pro) [reply tee-worker/litentry/core/vc-task/receiver/src/lib.rs]: The previous logic returns an error if the response is empty, but there is already code to process the empty response in `tee-worker/litentry/core/credentials/src/brc20/amount_holder.rs#line124~131`, So this VC logic is wrong, should return an empty/false VC here.

image

**Comment by Kailai-Wang** [comment]
Hi @higherordertech - just to confirm: this will **not** be a breaking change for the sdk or idh, right?

If so I guess it's ready to be merged

File: tee-worker/client-api/parachain-api/prepare-build/interfaces/sidechain/definitions.ts [hunk 5]
```diff
@@ -86,7 +86,6 @@ export default {
                 UnexpectedMessage: "Null",
                 __Unused_WrongSignatureType: "Null",
                 VerifyWeb3SignatureFailed: "Null",
-                NoEligibleIdentity: "Null",
```
Comment (wli-pro) [inline tee-worker/client-api/parachain-api/prepare-build/interfaces/sidechain/definitions.ts:89]: Should the corresponding line to be removed in client-sdk?

**Pull request merged**